Sources/Testing/Traits/Comment.swift

```diff
@@ -89,6 +89,41 @@ extension Comment: ExpressibleByStringLiteral, ExpressibleByStringInterpolation,
   }
 }
 
+// MARK: - ExpressibleByStringInterpolation
+
+extension Comment: ExpressibleByStringInterpolation {
+  public init(stringInterpolation: StringInterpolation) {
+    self.init(rawValue: stringInterpolation.rawValue)
+  }
+
+  /// The string interpolation handler type for ``Comment``.
+  @_documentation(visibility: private)
+  public struct StringInterpolation: StringInterpolationProtocol, Sendable {
+    /// Storage for the string constructed by this instance.
+    ///
+    /// `DefaultStringInterpolation` in the Swift standard library also simply
+    /// accumulates its result in a string.
+    @usableFromInline var rawValue: String = ""
+
+    @inlinable public init(literalCapacity: Int, interpolationCount: Int) {}
+
+    @inlinable public mutating func appendLiteral(_ literal: String) {
+      rawValue += literal
+    }
+
+    @inlinable public mutating func appendInterpolation(_ value: some Any) {
+      rawValue += String(describingForTest: value)
+    }
+
+    @inlinable public mutating func appendInterpolation(_ value: (some Any)?) {
+      // This overload is provided so that the compiler does not emit warnings
+      // about optional values in string interpolations (which we are fine with
+      // when constructing Comment instances.)
+      rawValue += String(describingForTest: value)
```

Comment:
Not sure about this, might be too magical and surprising compared to how string literal interpolation usually works. Is there a strong reason to include it?

Reply:
My expectation here (heh) is that if you pass an optional value in a testing context, you want to see its value. Also the warning's fix-it is emitted by the compiler and tells you to use `String(describing:)`, not `String(describingForTest:)`, which is wrong.

Reply:
I wouldn't say it's _wrong_ for the compiler to suggest that; users are welcome to call `String(describing:)` themselves. I think `String(describingForTest:)` is primarily for the testing library to call on values users provide, and (to a lesser extent) for use by test authors when writing helpers/utilities.

I think this violates the principle of least surprise. Regardless of what we think of the compiler's fix-it, users are generally accustomed to needing to handle optionals in string interpolations, and I don't know of a reason to deviate from that precedent here. It makes it seem as if the rules are more lax just because it's a test.

Reply:
The compiler emits that diagnostic because the default string representation of an optional value may either be `"Optional(x)"` (where `x` is the wrapped value's description) or `nil`, and neither of those values is likely expected by a developer who wants to just print `x`. In our case, that doesn't hold because optional values already have special handling in the library.

I would expect the following code to "just work" rather than emitting any diagnostics, for instance:

```swift
let x: Int? = f()
#expect(x == 10, "\(x) should be 10!")
```